blockchain-source/src/cardano/mod.rs

```diff
@@ -277,14 +278,15 @@ async fn block_fetch(
     let _ = block_fetcher.next().await?;
 
     while let Some(raw_block) = block_fetcher.next().await? {
-        let event = BlockEvent::from_serialized_block(raw_block.as_ref());
+        let event =
+            BlockEvent::from_serialized_block(raw_block.as_ref()).context(critical_error!());
 
         if channel
             .send(event.map(CardanoNetworkEvent::Block))
             .await
             .is_err()
         {
-            warn!("request response channel was closed, ignoring received blocks");
```

Comment:
Originally the idea here was to consume the range request entirely, so that the connection can still be used in the future, but currently if this happens it's because either:

The `CardanoSource` was dropped without calling `stop` first, so there is not really a way to recover once this happens, and it's quite noisy in the logs.

The `clear_buffers` function is called.

So this just changes it to an error, so that the connection is dropped early and we avoid processing a bunch of useless blocks. If a new request comes, then a new connection will be established in order to serve that anyway.